nntrainer/layers/layer_context.cpp

```diff
@@ -14,10 +14,20 @@
 #include <functional>
 
 #include <layer_context.h>
+#include <stdexcept>
 #include <var_grad.h>
 #include <weight.h>
 
 namespace nntrainer {
+void InitLayerContext::setOutputDimensions(
+  const std::vector<TensorDim> &out_dim) {
+  NNTR_THROW_IF(out_dim.size() < num_requested_out, std::invalid_argument)
```

Comment:
`out_dim.size() > num_requested_out` is allowed?

Reply:
Yes, that means rest will be dangled.

Reply:
Please include a warning in this case then, and provide a configuration option to make this hard error (the error part can be done later as well) as this can lead to bugs.

Reply:
I'll add warning for the later PR, this function is going through a bit of change.